### PR TITLE
switch on the `-pd` option for pw code

### LINKED ISSUE #1021: Issue with small systems on large machines
If I run a simple system (e.g. Lithium BCC or similar) with the fast protocol and 4 pools (for the pw.x part), when I get to the dos.x and projwfc.x parts, the calculations crash with 
```
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
     Error in routine  fft_type_set (6):
  there are processes with no planes. Use pencil decomposition (-pd .true.)
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```
Could we check if it's really just a matter of using this command line option, if it's safe to do in any case or when it should be used (maybe with an error handler?), or check anyway hot to ensure this crash does not happen?

This was run on Daint @ CSCS, with QE 7.1

### PR DESCRIPTION
**Pull request by superstar54**
fix #1021 
Enable pencil decomposition for all `pw.x` calculations.

**Note**, I still get issue on `projwfc`,
 
```console
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
     Error in routine  blk2cyc_zredist (1):
      nb less than the number of proc
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```

### CODE REVIEW

File: calculation_history.ipynb [hunk 1]
```diff
@@ -59,7 +59,7 @@
     "        </p>\n",
     "        <ul>\n",
     "            <li><b>Label Search Field:</b> Enter a job label to find matching jobs.</li>\n",
-    "            <li><b>Job State Dropdown:</b> Filter jobs based on their state (e.g., finished, waiting).</li>\n",
+    "            <li><b>Job State Dropdown:</b> Filter jobs based on their state (e.g., finished, running).</li>\n",
```
Comment (superstar54) [inline calculation_history.ipynb:62]: A small fix that is not related to this PR. But I fix it to avoid creating another PR.

File: src/aiidalab_qe/plugins/bands/workchain.py [hunk 2]
```diff
@@ -33,12 +34,18 @@ def update_resources(builder, codes):
     if "bands" in builder:
         set_component_resources(builder.bands.scf.pw, codes.get("pw"))
         set_component_resources(builder.bands.bands.pw, codes.get("pw"))
+        builder.bands.scf.pw.settings = orm.Dict({"CMDLINE": ["-pd", ".true."]})
```
Comment (AndresOrtegaGuerrero) [inline src/aiidalab_qe/plugins/bands/workchain.py:37]: should it be worth it to create a function in utils like use pencil_decomposition ? like that the code is not repeated , but also is easy to read in the future

Comment (superstar54) [reply src/aiidalab_qe/plugins/bands/workchain.py:37]: Good point, I will add a function.

Comment (AndresOrtegaGuerrero) [reply src/aiidalab_qe/plugins/bands/workchain.py]: perhaps also this should be control in the PwCode model ? so the user can set and unset this ?

Comment (superstar54) [reply src/aiidalab_qe/plugins/bands/workchain.py]: There are two ways to handle this:
- as you suggested, add a setting in the GUI so that the `advanced` user can control it.
- in the WorkChain, it handles the setting automatically or adds an error_handler to restart with a different number of CPUs.

I suggest discussing which way we want to implement in the future PR.

Comment (AndresOrtegaGuerrero) [reply src/aiidalab_qe/plugins/bands/workchain.py]: yeah , i think we can leave it here, but in another PR we just leave the freedom plus the explanation to the user

**Review by AndresOrtegaGuerrero — APPROVED** [review]
LGTM!, thank you @superstar54 , I would suggest you open an issue, for implementing this in the resource panel in such a way the users can activate or deactivate this. Plus adding explanation to the users about resources and the computers they are using

**Comment by superstar54** [comment]
Thanks @AndresOrtegaGuerrero , I opened an issue: https://github.com/aiidalab/aiidalab-qe/issues/1031
